lib/internal/priority_queue.js

```diff
@@ -23,7 +24,19 @@ module.exports = class PriorityQueue {
       this.#setPosition = setPosition;
   }
 
-  insert(value) {
+  updateSecondary() {
+    if (this.#size <= 1) {
+      this.#secondary = null;
+      return;
+    }
+
+    const top = this.peek();
+    this.removeAt(1, false);
+    this.#secondary = this.peek();
+    this.insert(top, false);
```

Comment:
hmmm, couldn't this be changed to taking the minimum between heap[1] and heap[2] instead of removing and inserting?

Reply:
If so, maybe the whole caching mechanism could be removed and instead just implement secondary to either return null if the size is <=1 or return the minimum between heap[2] and heap[3]?

Reply:
updated.